test/UnitTest.cs

```diff
@@ -3049,60 +3111,81 @@ <ScalarSubqueryExpr> 4
-                Output: bo.b2[0],bo.b1[1],b__2.b3[2],bo.b3[3]
-                Filter: bo.b1[1]=b__2.b1[4]
-                -> PhysicSingleJoin Left (actual rows=27, loops=3)
-                    Output: bo.b2[0],bo.b1[1],b__2.b3[3],bo.b3[2],b__2.b1[4]
-                    -> PhysicFromQuery <bo> (actual rows=9, loops=3)
-                        Output: bo.b2[1],bo.b1[0],bo.b3[2]
-                        -> PhysicNLJoin  (actual rows=9, loops=3)
-                            Output: b_2.b1[2],b_1.b2[0],b_1.b3[1]
-                            -> PhysicScanTable b as b_1 (actual rows=3, loops=3)
-                                Output: b_1.b2[1],b_1.b3[2]
-                            -> PhysicScanTable b as b_2 (actual rows=3, loops=9)
-                                Output: b_2.b1[0]
-                    -> PhysicScanTable b as b__2 (actual rows=3, loops=27)
-                        Output: b__2.b3[2],b__2.b1[0]
-                        Filter: b__2.b3[2]>1";
+                        Output: a.a1[2],a.a3[3],a.a2[4],b.b2[0]
+                        Filter: a.a1[2]=b.b1[1]
+                        -> PhysicScanTable b (actual rows=3)
+                            Output: b.b2[1],b.b1[0]
+                        -> PhysicScanTable a (actual rows=3)
+                            Output: a.a1[0],a.a3[2],a.a2[1]
+                -> PhysicFilter  (actual rows=3, loops=3)
+                    Output: bo.b2[0],b__4.b4[1],bo.b3[2],bo.b1[3]
+                    Filter: bo.b2[0]=b__4.b2[4]
+                    -> PhysicSingleJoin Left (actual rows=9, loops=3)
+                        Output: bo.b2[0],b__4.b4[3],bo.b3[1],bo.b1[2],b__4.b2[4]
+                        -> PhysicScanTable b as bo (actual rows=3, loops=3)
+                            Output: bo.b2[1],bo.b3[2],bo.b1[0]
+                        -> PhysicScanTable b as b__4 (actual rows=3, loops=9)
+                            Output: b__4.b4[3],b__4.b2[1]
+                            Filter: b__4.b3[2]>0
+        -> PhysicFilter  (actual rows=9, loops=3)
+            Output: b_2.b1[0],b__2.b3[1],b_1.b3[2],b_1.b2[3]
+            Filter: b_2.b1[0]=b__2.b1[4]
+            -> PhysicSingleJoin Left (actual rows=27, loops=3)
+                Output: b_2.b1[0],b__2.b3[3],b_1.b3[1],b_1.b2[2],b__2.b1[4]
+                -> PhysicNLJoin  (actual rows=9, loops=3)
+                    Output: b_2.b1[2],b_1.b3[0],b_1.b2[1]
+                    -> PhysicScanTable b as b_1 (actual rows=3, loops=3)
+                        Output: b_1.b3[2],b_1.b2[1]
+                        Filter: b_1.b2[1]<5
+                    -> PhysicScanTable b as b_2 (actual rows=3, loops=9)
+                        Output: b_2.b1[0]
+                -> PhysicScanTable b as b__2 (actual rows=3, loops=27)
+                    Output: b__2.b3[2],b__2.b1[0]
+                    Filter: b__2.b3[2]>1";
             TU.PlanAssertEqual(answer, phyplan);
-#endif
+        }
+
+        [TestMethod]
+        public void TestSubstrGroupBy()
```

Comment:
This does not have to be a separate test?

Reply:
I put it as a separate test earlier in a quick and dirty fashion but it belongs to either subquery removal or subquery rewrite. Since it exposes problem of column not being part of group by list, I moved it into the same group as TestFromQueryRemoval.